lib/util/historical-buffer-manager.ts

```diff
@@ -0,0 +1,95 @@
+/// <reference path='../../typings/tsd.d.ts' />
+
+import assert = require('assert');
+import interfaces = require('../interface');
+
+// Internal class that holds the buffer array, a counter for overflow items and maxBufferLength
+class Buffer<T> {
+    data: T[];
+    overflow: number;
```

Comment:
Just a preference, but you can also move the initialization here for syntactic sugar:
```typescript
data: T[] = [];
overflow: number = 0;
```